### PR TITLE
Fix reading bugs

### PR DESCRIPTION
**Pull request by kangpinghuang**
1. fix iterate scan range bug
2. fix read next block bug
3. fix filter empty rs_reader

### CODE REVIEW

File: be/src/olap/rowset/alpha_rowset_reader.cpp [hunk 1]
```diff
@@ -194,11 +192,21 @@ OLAPStatus AlphaRowsetReader::_get_next_row_for_singleton_rowset(RowCursor** row
         }
     }
     if (min_row == nullptr || min_index == -1) {
-        return OLAP_ERR_READER_READING_ERROR;
+        return OLAP_ERR_DATA_EOF;
     }
     *row = min_row;
     RowBlock* row_block = _row_blocks[min_index];
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_reader.cpp]: _row_blocks[min_index]->pos_inc() may be better

File: be/src/olap/rowset/alpha_rowset_reader.cpp [hunk 2]
```diff
@@ -207,6 +215,16 @@ OLAPStatus AlphaRowsetReader::_get_next_row_for_cumulative_rowset(RowCursor** ro
     OLAPStatus status = _get_next_not_filtered_row(pos, row);
     RowBlock* row_block = _row_blocks[pos];
     row_block->pos_inc();
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_reader.cpp]: _row_blocks[pos]->pos_inc() may be better

File: be/src/olap/rowset/alpha_rowset_reader.cpp [hunk 3]
```diff
@@ -154,19 +154,18 @@ OLAPStatus AlphaRowsetReader::_get_next_not_filtered_row(size_t pos, RowCursor**
             if (row_filtered) {
                 _current_read_context->stats->rows_del_filtered++;
                 current_row_block->pos_inc();
+                if (!current_row_block->has_remaining()) {
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_reader.cpp]: const DeleteHandler* delete_handler = _current_read_context->delete_handler;
    while (!found_row) {
        if (!current_row_block->has_remaining()) {
            OLAPStatus status = _get_next_block(pos, &current_row_block);
            if (status != OLAP_SUCCESS) {
                LOG(WARNING) << "_get_next_block failed, status:" << status;
                return status;
            }
        }

        size_t pos = current_row_block->pos();
        current_row_block->get_row(pos, *row);
        found_row = delete_handler->is_filter_data(_alpha_rowset_meta->version().second, *(*row));
    }

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/alpha_rowset_reader.cpp (hunk 1 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset_reader.cpp]: ok

File: be/src/olap/rowset/alpha_rowset_reader.cpp (hunk 2 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset_reader.cpp]: ok

File: be/src/olap/rowset/alpha_rowset_reader.cpp (hunk 3 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset_reader.cpp]: ok

**Pull request merged**